src/github.com/getlantern/balancer/balancer.go

```diff
@@ -49,7 +68,25 @@ func New(dialers ...*Dialer) *Balancer {
 // remaining Dialers until it either manages to connect, or runs out of dialers
 // in which case it returns an error.
 func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error) {
```

Comment:
This is the only function that returns a `net.Conn`, so it looks like a good place to check if the request is unencrypted and take action.